compiler/rustc_mir_transform/src/coverage/spans.rs

```diff
@@ -471,11 +473,16 @@ impl<'a> CoverageSpansGenerator<'a> {
                 previous iteration, or prev started a new disjoint span"
             );
             if dup.span.hi() <= self.curr().span.lo() {
-                let pending_dups = self.pending_dups.split_off(0);
-                for dup in pending_dups.into_iter() {
+                // Temporarily steal `pending_dups` into a local, so that we can
+                // drain it while calling other self methods.
+                let mut pending_dups = std::mem::take(&mut self.pending_dups);
+                for dup in pending_dups.drain(..) {
                     debug!("    ...adding at least one pending={:?}", dup);
                     self.push_refined_span(dup);
                 }
+                // The list of dups is now empty, but we can recycle its capacity.
+                assert!(pending_dups.is_empty() && self.pending_dups.is_empty());
+                self.pending_dups = pending_dups;
             } else {
                 self.pending_dups.clear();
```

Comment:
this is not necessary anymore, right? Could assert that it is empty instead of clearing it.

Reply:
When this clear occurs, `pending_dups` should always be *non*-empty. We checked that it's non-empty at the top of the function, and in this branch we haven't tried to drain it.

(I wanted to prove this by adding the assert and having it fail, but this line doesn't seem to ever be executed by the current coverage test suite.)

Reply:
This also makes me think that I should flatten the guard logic here too.

Reply:
> (I wanted to prove this by adding the assert and having it fail, but this line doesn't seem to ever be executed by the current coverage test suite.)

could assert it and wait for someone to report an issue with coverage 😆